File: zxlhhyccc/luci-app-nps-client-compat/luasrc/model/cbi/nps-client/nps_client.lua
m = Map("nps-client")
m.title = translate("Nps Client")
m.description = translate("Nps is a fast reverse proxy to help you expose a local server behind a NAT or firewall to the internet.")

m:section(SimpleSection).template = "nps-client/nps-client_status"

s = m:section(TypedSection, "nps-client")
s.addremove = false
s.anonymous = true

s:tab("setting", translate("Basic Setting"))
enable = s:taboption("setting", Flag, "enabled", translate("Enable"))
enable.rmempty = false

server = s:taboption("setting", Value, "server_addr", translate("Server"))
server.datatype = "host"
server.optional = false
server.rmempty = false

port = s:taboption("setting", Value, "server_port", translate("Port"))
port.datatype = "port"
port.default = "8024"
port.optional = false
port.rmempty = false

protocol = s:taboption("setting", ListValue, "protocol", translate("Protocol Type"))
protocol.default = "tcp"
protocol:value("tcp", translate("TCP Protocol"))
protocol:value("kcp", translate("KCP Protocol"))

auto_reconnection = s:taboption("setting", ListValue, "auto_reconnection", translate("Auto Reconnection"))
auto_reconnection.description = translate("Auto reconnect to the server when the connection is down.")
auto_reconnection.default = "true"
auto_reconnection:value("true", translate("True"))
auto_reconnection:value("false", translate("False"))

vkey = s:taboption("setting", Value, "vkey", translate("vkey"))
vkey.optional = false
vkey.password = true
vkey.rmempty = false

compress = s:taboption("setting", Flag, "compress", translate("Enable Compression"), translate("The contents will be compressed to speed up the traffic forwarding speed, but this will consume some additional cpu resources."))
compress.default = "1"
compress.rmempty = false

crypt = s:taboption("setting", Flag, "crypt", translate("Enable Encryption"), translate("Encrypted the communication between Npc and Nps, will effectively prevent the traffic intercepted."))
crypt.default = "1"
crypt.rmempty = false

log_level = s:taboption("setting", ListValue, "log_level", translate("Log Level"))
log_level:value(0, translate("Emergency"))
log_level:value(1, translate("Alert"))
log_level:value(2, translate("Critical"))
log_level:value(3, translate("Error"))
log_level:value(4, translate("Warning"))
log_level:value(5, translate("Notice"))
log_level:value(6, translate("Info"))
log_level:value(7, translate("Debug"))
log_level.default = "3"

return m

File: zxlhhyccc/luci-app-nps-client/htdocs/luci-static/resources/view/nps-client/nps-client.js
/* SPDX-License-Identifier: GPL-3.0-only
 *
 * Copyright (C) 2022 ImmortalWrt.org
 */

'use strict';
'require fs';
'require form';
'require view';
'require poll';
'require rpc';
'require ui';
'require uci';

var callServiceList = rpc.declare({
	object: 'service',
	method: 'list',
	params: ['name'],
	expect: { '': {} }
});

function getServiceStatus() {
	return L.resolveDefault(callServiceList('nps-client'), {}).then(function (res) {
		var isRunning = false;
		try {
			isRunning = res['nps-client']['instances']['instance1']['running'];
		} catch (e) { }
		return isRunning;
	});
}

function renderStatus(isRunning) {
	var renderHTML = "";
	var spanTemp = '<em><span style="color:%s"><strong>%s %s</strong></span></em>';

	if (isRunning) {
		renderHTML += String.format(spanTemp, 'green', _('Nps Client'),  _("RUNNING"));
	} else {
		renderHTML += String.format(spanTemp, 'red', _('Nps Client'), _("NOT RUNNING"));
	}

	return renderHTML;
}

return view.extend({
	render: function() {
		var m, s, o;

		m = new form.Map('nps-client', _('Nps Client'));
		m.description = _("Nps is a fast reverse proxy to help you expose a local server behind a NAT or firewall to the internet.");

		s = m.section(form.NamedSection, '_status');
		s.anonymous = true;
		s.render = function () {
			Poll.add(function () {
				return L.resolveDefault(getServiceStatus()).then(function(res) {
					var view = document.getElementById("service_status");
					view.innerHTML = renderStatus(res);
				});
			});

			return E('div', { class: 'cbi-map' },
				E('fieldset', { class: 'cbi-section'}, [
					E('p', { id: 'service_status' },
						_('Collecting data ...'))
				])
			);
		}

		s = m.section(form.TypedSection, 'nps-client', _('Settings'));
		s.tab('settings', _('Basic Settings'));
		s.anonymous = true;

		o = s.taboption('settings', form.Flag, 'enabled', _('Enable'));
		o.default = o.disabled;
		o.rmempty = false;

		o = s.taboption('settings', form.Value, 'server_addr', _('Server'));
		o.datatype = 'host';
		o.optional = false;
		o.rmempty = false;

		o = s.taboption('settings', form.Value, 'server_port', _('Port'));
		o.datatype = 'port';
		o.optional = false;
		o.rmempty = false;

		o = s.taboption('settings', form.ListValue, 'protocol', _('Protocol Type'));
		o.default = 'tcp';
		o.value('tcp', _('TCP Protocol'));
		o.value('kcp', _('KCP Protocol'));

		o = s.taboption('settings', form.ListValue, 'auto_reconnection', _('Auto Reconnection'),
			_('Auto reconnect to the server when the connection is down.'));
		o.default = 'true';
		o.value('true', _('True'));
		o.value('false', _('False'));

		o = s.taboption('settings', form.Value, 'vkey', _('vkey'));
		o.optional = false;
		o.password = true;
		o.rmempty = false;

		o = s.taboption('settings', form.Flag, 'compress', _('Enable Compression'), _('The contents will be compressed to speed up the traffic forwarding speed, but this will consume some additional cpu resources.'));
		o.default = '1';
		o.rmempty = false;

		o = s.taboption('settings', form.Flag, 'crypt', _('Enable Encryption'), _('Encrypted the communication between Npc and Nps, will effectively prevent the traffic intercepted.'));
		o.default = '1';
		o.rmempty = false;

		o = s.taboption('settings', form.ListValue, 'log_level', _('Log Level'));
		o.value('0', _('Emergency'));
		o.value('1', _('Alert'));
		o.value('2', _('Critical'));
		o.value('3', _('Error'));
		o.value('4', _('Warning'));
		o.value('5', _('Notice'));
		o.value('6', _('Info'));
		o.value('7', _('Debug'));
		o.default = '3';

		return m.render();
	}
});
